docs/guide/cursors.txt

```diff
@@ -1,3 +1,176 @@
 =======
 Cursors
 =======
+
+When a driver executes a read that returns mutliple documents, the server does not immediately return all values that match the query. Instead, the driver creates a **Cursor** object which fetches the documents in batches.
+
+Where are cursors used in the driver?
+-------------------------------------
+
+The following functions directly return cursors to the user:
+
+
+* ``Collection.prototype.find``
+* ``Collection.prototype.aggregate``
+* ``Collection.prototype.listIndexes``
+* ``Db.prototype.aggregate``
+* ``Db.prototype.listCollections``
+
+In addition, many other methods, like ``Collection.prototype.findOne`` and ``Collection.prototype.watch``\ , use cursors in order to return results.
+
+How do I use a cursor?
+----------------------
+
+There are multiple ways to consume a cursor:
+
+Stream API
+^^^^^^^^^^
+
+All cursors in the Node Driver are Node Readable Streams operating in Object Mode. Cursors will work with most Node stream APIs.
+
+.. code-block:: js
+
+   // defines a transform that calls JSON.stringify on the cursor's output
+   class StringifyStream extends stream.Transform {
```

Comment:
nit: you can also express this in simplified form:

```
const stringify = new Transform({
  objectMode: true,
  transform(chunk, _, callback) {
    callback(null, `${JSON.stringify(chunk)\n`);
  }
});
})
```

Reply:
I originally made it a class declaration b/c I was using it twice. Do you want me to inline it in both cases?

Reply:
I think you can define it above as an example of how to do this, and then reuse it in the code examples